<a href="https://colab.research.google.com/github/cesargar1507/DatasetsUB/blob/main/M3_AI3_GarciaCesar.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [22]:
library(readr)
library(tidyr)
library(dplyr)

ruta <- "https://raw.githubusercontent.com/cesargar1507/DatasetsUB/main/Facebook_10_18_raw.txt?_sm_au_=iVVW7DqFjJfSk0pM26q8vKHV6CTMc"
datos <- read.csv(ruta, sep = ";")
str(datos)
head(datos)

'data.frame':	468 obs. of  3 variables:
 $ pais     : chr  "Ukraine" "Kuwait" "Iraq" "Hong Kong SAR, China" ...
 $ valor    : chr  "23,3" "3100000" "2,5" "3408240" ...
 $ indicador: chr  "2010|Internet_por" "2018|Facebook" "2010|Internet_por" "2010|Facebook" ...


pais                 valor       indicador        
1 Ukraine              23,3        2010|Internet_por
2 Kuwait               3100000     2018|Facebook    
3 Iraq                 2,5         2010|Internet_por
4 Hong Kong SAR, China 3408240     2010|Facebook    
5 Ireland              1597000     2010|Facebook    
6 Kuwait               99,59999254 2018|Internet_por

In [23]:
datos_separados <- separate(datos, indicador, into = c("Año", "Variable"), sep = "\\|")

In [24]:
frec_año <- table(datos_separados$Año)
frec_año


2010 2018 
 234  234 

In [25]:
# Calcular las frecuencias de 'variable'
frec_variable <- table(datos_separados$Variable)
frec_variable


    Facebook Internet_por    poblacion 
         156          156          156 

In [26]:
# Crear un diccionario de mapeo para uniformizar los nombres de los países
diccionario_mapeo <- c(
  "Arab Rep. Egypt" = "Egypt",
  "Egypt, Arab Rep." = "Egypt",
  "Hong Kong SAR, China" = "Hong Kong",
  "Rep. Yemen" = "Yemen",
  "Russian" = "Russia",
  "Russian Federation" = "Russia",
  "Venezuela, RB" = "Venezuela",
  "Yemen, Rep." = "Yemen"
)

# Agregar una nueva columna "pais_uniforme" al data frame
datos_separados <- datos_separados %>%
  mutate(pais_r = ifelse(pais %in% names(diccionario_mapeo), diccionario_mapeo[trimws(pais)], pais))

In [27]:
frec_pais <- table(datos_separados$pais_r)
frec_pais


             Algeria            Argentina            Australia 
                   6                    6                    6 
             Austria         Bahamas, The              Bahrain 
                   6                    6                    6 
          Bangladesh              Belgium               Brazil 
                   6                    6                    6 
              Canada                Chile                China 
                   6                    6                    6 
            Colombia              Croatia              Denmark 
                   6                    6                    6 
               Egypt              Estonia             Ethiopia 
                   6                    6                    6 
             Finland               France              Germany 
                   6                    6                    6 
              Greece            Hong Kong              Hungary 
                   6                   

In [28]:
# Pivote de la tabla a formato wide
tabla_wide <- pivot_wider(data = datos_separados, id_cols= c(pais_r,Año), names_from = "Variable", values_from = "valor")
tabla_wide <- arrange(tabla_wide, pais_r)
tabla_wide

pais_r               Año  Internet_por Facebook  poblacion    
1   Algeria              2010 12,5         928100      35977      
2   Algeria              2018 59,57968476  19000000    42228      
3   Argentina            2018 78           30000000    44494      
4   Argentina            2010 45           10542040    40788      
5   Australia            2010 76           9009660     22031      
6   Australia            2018 88           15000000    24992      
7   Austria              2018 87,71046495  3700000     8847       
8   Austria              2010 75,17        2084840     8363       
9   Bahamas, The         2010 43           139460      354        
10  Bahamas, The         2018 83,4         220000      385        
11  Bahrain              2018 98,64342901  1100000     1569       
12  Bahrain              2010 55           226480      1240       
13  Bangladesh           2010 3,7          854520      147575     
14  Bangladesh           2018 57,2         28000000    161356     
15  Belgium              2010 75           3505920     10895      
16  Belgium              2018 88,65506428  6500000     11422      
17  Brazil               2018 66           139000000   209469     
18  Brazil               2010 40,65        4757200     195713     
19  Canada               2010 80,3         15497900    34004      
20  Canada               2018 90           23000000    37058      
21  Chile                2018 77,5         13000000    18729      
22  Chile                2010 45           6944540     17062      
23  China                2010 34,3         119480      1337705    
24  China                2018 53           1800000     1392730    
25  Colombia             2018 63,2         29000000    49648      
26  Colombia             2010 36,5         10226820    45222      
27  Croatia              2010 56,55        1192160     4295       
28  Croatia              2018 72,68877088  1800000     4089       
29  Denmark              2018 97,64426813  3700000     5797       
30  Denmark              2010 88,72        2527000     5547       
⋮   ⋮                    ⋮    ⋮            ⋮         ⋮            
127 Sweden               2018 92,14167705  6300000     10183      
128 Sweden               2010 90           3798020     9378       
129 Switzerland          2010 83,9         2234380     7824       
130 Switzerland          2018 93,7         3700000     8516       
131 Taiwan               2010 NA           6745160     23.140,95  
132 Taiwan               2018 88           18000000    23.580,08  
133 Thailand             2018 56,81748093  46000000    69428      
134 Thailand             2010 22,4         4216760     67195      
135 Tunisia              2018 64,19081023  6400000     11565      
136 Tunisia              2010 36,8         1594700     10635      
137 Turkey               2018 71,04276107  44000000    82319      
138 Turkey               2010 39,82        22552540    72326      
139 Uganda               2018 40,5         2600000     42723      
140 Uganda               2010 12,5         168720      32428      
141 Ukraine              2010 23,3         425680      45870      
142 Ukraine              2018 93,4         9500000     44622      
143 United Arab Emirates 2010 68           1596160     8549       
144 United Arab Emirates 2018 98,45000178  8700000     9630       
145 United Kingdom       2010 85           26543600    62766      
146 United Kingdom       2018 94,89674176  44000000    66488      
147 United States        2010 71,69        125881220   309326     
148 United States        2018 88           240000000   327167     
149 Uzbekistan           2010 15,9         26840       28562      
150 Uzbekistan           2018 52,3         800000      32955      
151 Venezuela            2018 53,1         13000000    28870      
152 Venezuela            2010 37,37        6686300     28439      
153 Vietnam              2010 30,65        1019120     87967      
154 Vietnam              2018 70,34963558  50000000

In [29]:
tabla_wide$Facebook <- as.numeric(tabla_wide$Facebook)
tabla_wide$Internet_por <- gsub(",",".",tabla_wide$Internet_por)
tabla_wide$Internet_por <- as.numeric(tabla_wide$Internet_por)
tabla_wide$poblacion <- gsub(",","",tabla_wide$poblacion)
tabla_wide$poblacion <- gsub("\\.","",tabla_wide$poblacion)
tabla_wide$poblacion <- as.numeric(tabla_wide$poblacion)
summary(select(tabla_wide, Facebook, Internet_por, poblacion))

    Facebook          Internet_por     poblacion      
 Min.   :    26840   Min.   : 0.75   Min.   :    318  
 1st Qu.:  1402950   1st Qu.:40.58   1st Qu.:   6806  
 Median :  3543690   Median :66.00   Median :  24073  
 Mean   : 15175281   Mean   :60.80   Mean   : 108199  
 3rd Qu.: 14150075   3rd Qu.:83.55   3rd Qu.:  66613  
 Max.   :300000000   Max.   :99.65   Max.   :2358008  
                     NA's   :1                        

In [30]:
datos_spain <- filter(tabla_wide, pais_r == "Spain")
datos_spain

pais_r Año  Internet_por Facebook poblacion
1 Spain  2018 86.10724     23000000 46723    
2 Spain  2010 65.80000     10610080 46576

In [31]:
incremento_face = ((filter(datos_spain,Año=="2018")$Facebook-filter(datos_spain,Año=="2010")$Facebook)/filter(datos_spain,Año=="2010")$Facebook)*100
paste(incremento_face,"%")

[1] "116.774991329 %"

En primer lugar se detecta la inconsistencia en cuanto a la población total de España, no es posible que sean solamente 46,000 habitantes, dejando de lado esto, se identifica un incremento en cuanto al uso de internet por parte de los habitantes de este país de un 65% a un 86% de la población, además, se identifica un incremento de usuarios de facebook de un 116% del año 2010 al 2018